packages/web3-core-method/src/index.js

```diff
@@ -314,13 +314,35 @@ Method.prototype._confirmTransaction = function (defer, result, payload, extraFo
             utils._fireError({message: 'Failed to subscribe to new newBlockHeaders to confirm the transactions receipt. Are you using HttpProvider? Please switch to Websockets.', data: err}, defer.eventEmitter, defer.reject);
         }
     };
-
+    
+  // start watching for confirmation depending on the support features of the provider
+  var startWatching = function() {
     // if provider allows PUB/SUB
     if (_.isFunction(this.requestManager.provider.on)) {
-        method.eth.subscribe('newBlockHeaders', checkConfirmation);
+      method.eth.subscribe('newBlockHeaders', checkConfirmation);
     } else {
-        intervalId = setInterval(checkConfirmation, 1000);
+      intervalId = setInterval(checkConfirmation, 1000);
     }
+  };
+
+  // first check if we already have a confirmed transaction
+  method.eth.getTransactionReceipt(result)
+    .then(function(receipt) {
+      if (receipt && receipt.blockNumber) {
+        checkConfirmation(null, receipt.blockNumber, null, receipt);
+        setTimeout(function(){
```

Comment:
Yes I was wondering, if we would still need it, but the only reason, we should still start watching, even after we have received the receipt, may be to handle the `"confirmation"`event in the promievent, so we can count the blocks. But I changed it, so we only continue to watch, if there are eventhandlers registered. But for the InstantSealEngine this does not really make sense.